src/utils/BucketUtils.ts

```diff
@@ -102,6 +112,8 @@ export const binCollectionToBuckets = async (collection: DarCollection, dacIds:
     const dacs: DacTerm[] = terms
       .filter((term: DatasetTerm) => dacIds.includes(term.dacId))
       .map((term: DatasetTerm) => term.dac)
+      .filter((dac: DacTerm | undefined) => !isNil(dac))
+      .filter((dac: DacTerm, index: number, self: DacTerm[]) => self.findIndex((d: DacTerm) => d.dacId === dac.dacId) === index)
```

Comment:
This is the only functional change to this file, all other changes address sonar warnings. This filters the DACs in a bucket to unique values. It is common for several datasets in a bucket to all belong to the same DAC and that resulted in duplicate values here.